crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -263,16 +274,26 @@ async fn download_block(
                     .with_context(move || format!("Verify block {}", block_number))?;
                 Ok((block, verify_result))
             });
-            #[allow(unused_variables)]
             let (block, verify_result) = verify_hash.await.context("Verify block hash")??;
-            // FIXME: test block hashes aren't correct so this error breaks tests.
-            #[cfg(not(test))]
-            anyhow::ensure!(
-                verify_result != crate::state::block_hash::VerifyResult::Mismatch,
-                "Block hash mismatch"
-            );
-            match block.status {
-                Status::AcceptedOnL1 | Status::AcceptedOnL2 => Ok(DownloadBlock::Block(block)),
+            match (block.status, verify_result) {
+                (Status::AcceptedOnL1 | Status::AcceptedOnL2, VerifyResult::Match(commitments)) => {
+                    Ok(DownloadBlock::Block(block, commitments))
+                }
+                (Status::AcceptedOnL1 | Status::AcceptedOnL2, VerifyResult::NotVerifiable) => {
+                    Ok(DownloadBlock::Block(block, Default::default()))
+                }
+
+                #[cfg(test)]
+                // FIXME: test block hashes aren't correct so `VerifyResult::Mismatch` breaks tests.
```

Comment:
But why wasn't this behavior causing tests to fail previously? (I'm just wondering why the old `anyhow::ensure!()` behavior wasn't triggering the same issue.)

Reply:
The previous test failures were caused by me messing up with `mainnet.sqlite` from the test fixtures, when I restored it the test started to pass. I don't think this workaround cause any issues with tests. But let me double-check locally.

Reply:
Hmm, so do we still need these `#[cfg(test)]` conditions here then? (Previously tests passed with effectively what's the `not(test)` case now.)

Reply:
Previously it was "if cfg(test) continue even if there is a mismatch". Now it is "if cfg(test) and mismatch - return OK with block and empty commitments". Without `#[cfg(not(test))]`-case when block hash is mismatched the error turns from `Block hash mismatch` into `Rejecting block as its status is ...`, failing the test that seems to check this exact case.

I want to "make it right" (drop conditional compilation) in a follow-up https://github.com/eqlabs/pathfinder/pull/835.

Reply:
Ah, my bad -- you're right. Thanks for the explanation, I didn't notice the `#[cfg(not(test))]` in the old code... :facepalm: